src/net.cpp

```diff
@@ -1406,6 +1429,16 @@ void CConnman::ThreadSocketHandler()
     }
 }
 
+void CConnman::SetMaxConnections(int newMaxConnections)
+{
+    newMaxConnections = std::max(newMaxConnections, MAX_ADDNODE_CONNECTIONS);
```

Comment:
I think we need to do the `std::max` against `MAX_ADDNODE_CONNECTIONS + PROTECTED_INBOUND_PEERS` here: when I `setmaxconnections` below `28` but above `MAX_ADDNODE_CONNECTIONS ` (eg: `10`), I get into a loop trying to get under 28 but it of course never gets there and just keeps trying...

```
2022-01-22 21:28:45 ThreadSocketHandler: attempting to reduce connections: max=10 current=28
2022-01-22 21:28:45 ThreadSocketHandler: attempting to reduce connections: max=10 current=28
2022-01-22 21:28:45 ThreadSocketHandler: attempting to reduce connections: max=10 current=28
```

Reply:
That's reasonable. I considered two alternatives here. Your suggestion is technically accurate and means that the user is setting the maximum number of outbound connections allowed. The approach I took is intended to allow the user to set the maximum number of inbound plus outbound connections allowed.

I struggled with trying to explain the nuances of your approach in the RPC documentation and whether and how to expose the value of `PROTECTED_INBOUND_PEERS` to the UI. I have a small concern that someone would use this command, then check the number of connections with `getconnectioncount` and worry about the discrepancy.

I think there's a way to explain it without going into too much detail though.

Reply:
> I have a small concern that someone would use this command, then check the number of connections with getconnectioncount and worry about the discrepancy.

One of the things I was thinking for followup was maybe expose `maxconnections` on the `getnetworkinfo` RPC response... I didn't want to make that blocking for this discussion. Could do the same on UI then.

Reply:
I like that idea. Let's do it in a followup PR?